### PR TITLE
Move to Puppet 4

### PR DESCRIPTION
**Pull request by JayH5**
* Ability to add arbitrary "accept" rules.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,14 @@ To allow access to Docker containers you would do:
 -A DOCKER_INPUT -s 192.168.0.1/32 -j DOCKER
 ```
 
+The `$accept_rules` parameter for the main `docker_firewall` class provides an easy way to set up rules to accept connections to Docker containers. This parameter takes a hash of `firewall` resources to create, and defaults those rules to be in the `DOCKER_INPUT` chain and jump to the `DOCKER` chain. This parameter is probably best set using Hiera, for example:
+```yaml
+docker_firewall::accept_rules:
+  200 accept all traffic from 192.168.0.1/32:
```
Comment (jerith) [inline README.md:56]: Should this YAML key be quoted?

File: metadata.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
   "requirements": [
     {
       "name": "puppet",
-      "version_requirement": ">= 3.4.0"
+      "version_requirement": ">= 4.5.0"
```
Comment (jerith) [inline metadata.json:23]: Shouldn't we bump the module version as well? We usually do that at release time, but I think the new incompatibility with Puppet 3 makes it worth doing in this branch.

File: manifests/init.pp [hunk 3]
```diff
@@ -55,31 +55,29 @@
...
+  Optional[String]               $prerouting_nat_policy        = undef,
+  Variant[String, Array[String]] $output_nat_purge_ignore      = [],
+  Optional[String]               $output_nat_policy            = undef,
+  Variant[String, Array[String]] $postrouting_nat_purge_ignore = [],
+  Optional[String]               $postrouting_nat_policy       = undef,
+  Variant[String, Array[String]] $forward_filter_purge_ignore  = [],
+  Optional[String]               $forward_filter_policy        = 'drop',
+
+  Boolean            $accept_eth0  = false,
+  Boolean            $accept_eth1  = false,
```
Comment (jerith) [inline manifests/init.pp]: Should this maybe be a list of device names to accept all traffic from rather than hardcoding just two devices?

**Review by jerith — COMMENTED** [review]
Looks good so far.

File: metadata.json (hunk 2 above)
Comment (JayH5) [reply metadata.json:23]: I already bumped the version to `0.2.0-dev` in a previous PR but never released a 0.2.0.

Comment (jerith) [reply metadata.json:23]: Cool.

File: manifests/init.pp (hunk 3 above)
Comment (JayH5) [reply manifests/init.pp]: Yeah I was thinking of just removing these two parameters completely and relying on `$accept_rules`. But was trying not to get carried away in this PR.

File: README.md (hunk 1 above)
Comment (JayH5) [reply README.md:56]: Works fine like this :)

Comment (jerith) [reply README.md:56]: YAML quoting rules confuse me, so I generally quote strings fairly often. I'm happy with this if it works, though.

File: manifests/init.pp (hunk 3 above)
Comment (jerith) [reply manifests/init.pp]: I think let's remove these here, since we're breaking compat by moving to Puppet 4 anyway.

**Review by jerith — APPROVED** [review]
:+1:

**Pull request merged**